barretenberg/cpp/src/barretenberg/api/bbrpc_execute.hpp

```diff
@@ -0,0 +1,219 @@
+{
+    return response.visit([](const auto& resp) -> const std::string& { return resp.error_message; });
+}
+
+inline CircuitProve::Response execute(BB_UNUSED BBRpcRequest& request, CircuitProve&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline CircuitComputeVk::Response execute(BB_UNUSED BBRpcRequest& request, CircuitComputeVk&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline CircuitVerify::Response execute(BB_UNUSED BBRpcRequest& request, CircuitVerify&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline CircuitInfo::Response execute(BB_UNUSED BBRpcRequest& request, CircuitInfo&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline CircuitCheck::Response execute(BB_UNUSED BBRpcRequest& request, CircuitCheck&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline ProofAsFields::Response execute(BB_UNUSED BBRpcRequest& request, ProofAsFields&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline VkAsFields::Response execute(BB_UNUSED BBRpcRequest& request, VkAsFields&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline ClientIvcStart::Response execute(BBRpcRequest& request, BB_UNUSED ClientIvcStart&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline ClientIvcLoad::Response execute(BBRpcRequest& request, ClientIvcLoad&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline ClientIvcAccumulate::Response execute(BBRpcRequest& request, ClientIvcAccumulate&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline ClientIvcProve::Response execute(BBRpcRequest& request, ClientIvcProve&& command)
+{
+    (void)request;
+    (void)command;
+    throw_or_abort("code in progress! should not be called");
+}
+
+inline std::shared_ptr<ClientIVC::DeciderProvingKey> get_acir_program_decider_proving_key(
+    const BBRpcRequest& request, acir_format::AcirProgram& program)
+{
+    ClientIVC::ClientCircuit builder = acir_format::create_circuit<ClientIVC::ClientCircuit>(program);
+
+    // Construct the verification key via the prover-constructed proving key with the proper trace settings
+    return std::make_shared<ClientIVC::DeciderProvingKey>(builder, request.trace_settings);
+}
+
+inline ClientIVC::VerificationKey compute_vk_for_ivc(const BBRpcRequest& request,
```

Comment:
From our discussion this morning: once the hiding circuit is in noir, this method can *utilize* the "standalone" vk method on the hiding circuit bytecode, but we do still need a separate method since the CIVC VK is {mega_vk (hiding), eccvm VK, translator VK}